File: nbui-core/src/main/java/cn/rubintry/nbui/core/utils/ScreenUtils.java
package cn.rubintry.nbui.core.utils;

import android.content.Context;
import android.graphics.Point;
import android.view.WindowManager;

public class ScreenUtils {

    public static int getScreenHeight(Context context){
        WindowManager wm = (WindowManager)context.getSystemService(Context.WINDOW_SERVICE);
        if (wm == null) {
            return -1;
        }
        Point point = new Point();
        wm.getDefaultDisplay().getRealSize(point);
        return point.y;
    }
}

File: nbui-pull/src/main/java/cn/rubintry/nbui/pull/ScreenUtils.kt
package cn.rubintry.nbui.pull

import android.content.Context
import android.graphics.Point
import android.os.Build
import android.view.WindowManager

/**
 * @author rubintry
 */
object ScreenUtils {
    fun getScreenHeight(context: Context): Int {
        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
            ?: return -1
        val point = Point()
        wm.defaultDisplay.getRealSize(point)
        return point.y
    }

    fun getScreenWidth(context: Context): Int {
        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
            ?: return -1
        val point = Point()
        wm.defaultDisplay.getRealSize(point)
        return point.x
    }
}